### PR TITLE
Generate work for document

### PR DESCRIPTION
**Pull request by quang-ng**
## Context
Enhancements to PDF Upload Process
- Refactored upload_pdfs.py: Improved the efficiency and readability of the PDF upload logic.
- Updated run_upload_pdf_to_s3.py: Streamlined the script execution for uploading PDFs to S3.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -34,16 +34,25 @@ jobs:
       - name: Install dependencies
         run: |
           pip install uv
-          uv pip install --system -r <(uv pip compile --all-extras pyproject.toml)
+          uv sync --all-extras
+          pip install .[ci]
```
Comment (leej3) [inline .github/workflows/tests.yml:38]: this isn't required with --all-extras

Comment (leej3) [inline .github/workflows/tests.yml:37]: this isn't required with all --all-extras

File: dsst_etl/_utils.py [hunk 2]
```diff
@@ -6,7 +6,7 @@ def get_compute_context_id():
 
 
 def get_bucket_name():
-    bucket_name = os.getenv('S3_BUCKET_NAME')
+    bucket_name = os.getenv("S3_BUCKET_NAME")
```
Comment (leej3) [inline dsst_etl/_utils.py]: this should make use of the config object

File: dsst_etl/upload_pdfs.py [hunk 3]
```diff
@@ -61,94 +62,103 @@ def upload_pdfs(self, pdf_paths: List[str]) -> Tuple[List[str], List[str]]:
...
         Returns:
             Provenance: Created provenance record
         """
         provenance = Provenance(
             pipeline_name="Document Upload",
             version=__version__,
             compute=get_compute_context_id(),
-            personnel=os.environ.get('HOSTNAME'),
-            comment=comment
+            personnel=os.environ.get("HOSTNAME"),
```
Comment (leej3) [inline dsst_etl/upload_pdfs.py]: config object too

File: dsst_etl/upload_pdfs.py [hunk 4]
```diff
@@ -157,51 +167,47 @@ def link_documents_to_work(
...
-        os.path.join(pdf_directory_path, f) 
-        for f in os.listdir(pdf_directory_path) 
-        if f.lower().endswith('.pdf')
-    ]
-    
+    # Convert string path to Path object
+    pdf_directory = Path(pdf_directory_path)
+
+    # Get list of PDF files using glob
+    pdf_files = [str(pdf_file) for pdf_file in pdf_directory.glob("*.pdf")]
```
Comment (leej3) [inline dsst_etl/upload_pdfs.py]: I think this could be  `list(pdf_directory.glob("*.pdf"))`

**Review by leej3 — APPROVED** [review]
Thanks, this looks great. Some small nitpicks

File: tests/test_pdf_uploader.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+from unittest.mock import patch, MagicMock
+from dsst_etl.upload_pdfs import PDFUploader
+from dsst_etl.db import get_db_session
+from dsst_etl.models import Documents, Provenance, Works
+from pathlib import Path
+
+class TestPDFUploader(unittest.TestCase):
+
+    @patch('dsst_etl.upload_pdfs.boto3.client')
+    def setUp(self, mock_boto_client):
```
Comment (joshlawrimore) [inline tests/test_pdf_uploader.py:15]: I needed to manually create the database by running db_init() before these test would pass. Please added init_db() to the unittest.

File: tests/test_pdf_uploader.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        mock_boto_client.return_value = self.mock_s3_client
+
+        # Start a new database session and transaction
+        self.session = get_db_session()
+        self.session.begin_nested()  # Start a nested transaction
+
+        # Initialize PDFUploader with the session
+        self.uploader = PDFUploader(self.session)
+
+    def tearDown(self):
```
Comment (joshlawrimore) [inline tests/test_pdf_uploader.py:30]: After running this test, I still have records left in my Works, Documents, and Provenance tables. Please check this so the test does not leave records.

Comment (joshlawrimore) [reply tests/test_pdf_uploader.py:30]: This appears fixed in the latest version. Thanks!

**Pull request merged**